## 1. State the Hypotheses Statement:
### To investigate the restaurant owners' claim about the increase in weekly operating costs, we need to formulate the null and alternative hypotheses:

### Null Hypothesis (H₀): The theoretical weekly operating cost model is accurate.

### H₀ : u = 1000 + 5X

### Alternative Hypothesis (H₁): The theoretical weekly operating cost model is not accurate, and the actual weekly operating costs are higher.
    
### H1 : u > 1000 + 5X

## 2. Calculate the Test Statistic:

In [2]:
import scipy.stats as stats
import numpy as np

# Given Data
sample_mean = 3050 # sample mean weekly cost
X_mean = 600 # mean units produced per week
X_std = 25 # standard deviation of units produced per weeek
n = 25 # sample size

# Calculating the mean of weekly cost
W_mean = 1000 + 5 * X_mean
print("Sample_mean:",sample_mean,"\nX_mean:", X_mean,"\nX_std:",X_std,"\nSample Size(n):",n,"\nW_mean:",W_mean)

Sample_mean: 3050 
X_mean: 600 
X_std: 25 
Sample Size(n): 25 
W_mean: 4000


In [3]:
# Calculate the standard deviation of the weekly operating costs
cost_std = 5 * X_std
print("cost_std:",cost_std)

cost_std: 125


In [4]:
# Calculate the Test Statistic
test_stat = (sample_mean-W_mean) / (cost_std / np.sqrt(n))
print("Test_Statistic:",test_stat)

Test_Statistic: -38.0


## 3. Determine the Critical Value:

In [8]:
# Determine the critical value for a one-tailed test at alpha = 0.05
alpha = 0.05
critical_value = -stats.norm.ppf(1 - alpha)
print("Critical_value:",critical_value)

Critical_value: -1.6448536269514722


In [11]:
if test_stat < critical_value:
    print("Reject the null hypothesis: There is strong evidence to support the claim that the weekly operating costs are higher than the model suggests.")
else:
    print("Fail to reject the null hypothesis: There is not enough evidence to support the claim that the weekly operating costs are higher than the model suggests.")

Reject the null hypothesis: There is strong evidence to support the claim that the weekly operating costs are higher than the model suggests.


## Compare the test statistic with the critical value:

### If t < -1.645, we reject the null hypothesis.
### Since 𝑡 = −38
### which is much less than -1.645, we reject the null hypothesis.

## 5. Conclusion:
### Based on the decision in step 4, there is strong evidence to support the restaurant owners
### ' claim that the weekly operating costs are higher than the model suggests.

# Chisquare Test

## 1.Statement of Hypothesis
### H₀: There is no significant association between the type of smart home device purchased and the customer satisfaction level.
### H₁: There is a significant association between the type of smart home device purchased and the customer satisfaction level.

In [17]:
import numpy as np
import scipy.stats as stats

# Observed Frequencies
observed = np.array([
    [50,70],
    [80,100],
    [60,90],
    [30,50],
    [20,50]])
observed

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [20]:
# Margin total
row_total =  np.sum(observed, axis =1)
col_total = np.sum(observed, axis = 0)
Total = np.sum(observed)

print("Row-Total:",row_total,"\nCol-Total:",col_total,"\nTotal:",Total)

Row-Total: [120 180 150  80  70] 
Col-Total: [240 360] 
Total: 600


In [21]:
# Expected frequencies
expected = np.outer(row_total, col_total) / Total
expected

array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]])

## 2. Compute the Chi-Square Statistic:

In [22]:
# Chi-Square Statistic

chi_square_test = np.sum((observed - expected)**2 / expected)

print("Chi_Square_Value:",chi_square_test)

Chi_Square_Value: 5.638227513227513


## 3. Determine the Critical Value:

In [24]:
# Degree of Freedom 
df = (observed.shape[0]-1) * (observed.shape[1] -1)

# Critical value at 0.05 significance level
alpha = 0.05
critical_value = stats.chi2.ppf(1-alpha,df)

print("Degree Of Freedom:",df,"\nCritical_value:",critical_value)

Degree Of Freedom: 4 
Critical_value: 9.487729036781154


## Decision

### Since the Chi-Square statistic (5.638) is less than the critical value (9.488), we fail to reject the null hypothesis.